dll/win32/shell32/droptargets/CFSDropTarget.cpp

```diff
@@ -83,12 +83,26 @@ HRESULT CFSDropTarget::_CopyItems(IShellFolder * pSFFrom, UINT cidl,
     if (!pszSrcList)
         return E_OUTOFMEMORY;
 
+    BOOL bRenameOnCollision = FALSE;
+    if (bCopy)
+    {
+        WCHAR szPath1[MAX_PATH], szPath2[MAX_PATH];
+        lstrcpynW(szPath1, pszSrcList, _countof(szPath1));
+        PathRemoveFileSpecW(szPath1);
+        lstrcpynW(szPath2, m_sPathTarget, _countof(szPath2));
+        PathRemoveFileSpecW(szPath2);
+        if (lstrcmpiW(szPath1, szPath2) == 0)
```

Comment:
Oh i misread the code, sorry. You are _**keeping**_ the directories in the paths, not the file names. Please disregard my above suggestion. It may be possible to find where the file name specification currently starts in both paths (keeping those `PathFindFileNameW` calls), and do the string comparison for the substring that starts from the beginning and up to the file part (using lstrncmpiW or sthg):

```suggestion
        PCWSTR pszFileName1 = PathFindFileNameW(pszSrcList);
        PCWSTR pszFileName2 = PathFindFileNameW(m_sPathTarget);
        SIZE_T cchPath1 = pszFileName1 - pszSrcList;
        SIZE_T cchPath2 = pszFileName2 - m_sPathTarget;

        if ((cchPath1 == cchPath2) && lstrncmpiW(szPath1, szPath2) == 0)
```
_**(NOTE: This would require to copy-paste this helper function from https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/comctl32/listview.c;hb=0e8cf6ffd58cc7a46db8bfe46cfb361aceb15153#l545  since lstrncmpiW is not an existing export... -- That is, if you want to really use those `lstr*W` functions. Otherwise, using `_wcsnicmp` might be sufficient.)**_

Also, I've the separate question: Are the paths, at this stage, already normalized, i.e. that for example:
`"C:\Some\Path\..\where\file.txt"` has already become `"C:\Some\where\file.txt"` ?

Reply:
> Also, I've the separate question: Are the paths, at this stage, already normalized, i.e. that for example: "C:\Some\Path\..\where\file.txt" has already become "C:\Some\where\file.txt" ?

```txt
CDefaultContextMenu::InvokeCommand
    --> CDefaultContextMenu::DoPaste
        --> shlwapi!SHSimulateDrop
            --> CFSDropTarget::Drop
                --> CFSDropTarget::_DoDrop
                    --> CFSDropTarget::_CopyItems
```

`DoPaste` comes from clipboard. The PIDL of `DoPaste` can be a relative PIDL.